### PR TITLE
Delete v2_user_routes and associated files

### PR DESCRIPTION
**Pull request by maddiedierker**
Now that we have removed the pegasus teacher dashboard, we can safely remove the following:
- v2_user_routes minus `/v2/user` endpoint [used by NetSim](https://github.com/code-dot-org/code-dot-org/blob/1cad5ac667f9ed1a3aa31063e6d06bd21a78e2bd/apps/src/netsim/DashboardUser.js#L75) (+ unit tests)
- teacher_dashboard_helpers
- section_api_helpers (+ unit tests)
- account-management-api.md

Also deleting a test that was checking for `text/ng-template` responses from the pegasus teacher dashboard templates, which have been removed (not sure how this test wasn't failing from #27884, where I removed these templates).

Thanks to Dave and Suresh for calling these out!

### CODE REVIEW

File: pegasus/routes/v2_user_routes.rb [hunk 1]
```diff
@@ -1,45 +0,0 @@
-require 'honeybadger/ruby'
-
-get '/v2/user' do
```
Comment (davidsbailey) [inline pegasus/routes/v2_user_routes.rb]: I did find one reference to this route: https://github.com/code-dot-org/code-dot-org/blob/1cad5ac667f9ed1a3aa31063e6d06bd21a78e2bd/apps/src/netsim/DashboardUser.js#L75

perhaps possible to preserve `/v2/user` and still delete most everything else in this PR?

**Review by davidsbailey — COMMENTED** [review]
Nice work Maddie! It looks like https://github.com/code-dot-org/code-dot-org/blob/staging/docs/account-management-api.md can be removed now also.

I can take a look again after /v2/user is addressed.

**Comment by sureshc** [comment]
Thanks very much!  Looks like there are a couple of valid uses for the Pegasus `fake_dashboard` remaining even after this Pull Request, so I may need to stub some Scenic functionality anyway.

File: pegasus/routes/v2_user_routes.rb (hunk 1 above)
Comment (maddiedierker) [reply pegasus/routes/v2_user_routes.rb]: 😱  great catch!

**Review by davidsbailey — APPROVED** [review]
Looking great! Screen Shot 2019-04-18 at 12 33 37 PM ❤️❤️❤️

**Comment by sureshc** [comment]
Looks like there are still some dependencies on some of the deleted files (such as `section_api_helpers.rb`): https://app.honeybadger.io/projects/45435/faults/48480779

**Pull request merged**